src/main/java/org/dswarm/xsd2jsonschema/JsonSchemaParser.java

```diff
@@ -174,31 +175,35 @@ else if (term instanceof XSWildcard) {
 			final XSTerm term = xsParticle.getTerm();
 			if (term instanceof XSModelGroup) {
 
-				list.addAll(iterateParticle(xsParticle));
+				list.addAll(iterateParticle(xsParticle,depth+1));
 			}
 
 			else if (term instanceof XSModelGroupDefinition) {
 				final XSModelGroupDefinition xsModelGroupDefinition = (XSModelGroupDefinition) term;
-				list.addAll(iterateModelGroup(xsModelGroupDefinition.getModelGroup()));
+				list.addAll(iterateModelGroup(xsModelGroupDefinition.getModelGroup(),depth + 1 ));
 			}
 
 			else {
 
-				final Optional<JSElement> optionalJSElement = iterateSingleParticle(xsParticle);
+				final Optional<JSElement> optionalJSElement = iterateSingleParticle(xsParticle, depth+1);
 				optionalJSElement.ifPresent(list::add);
 			}
 		}
 
 		return list;
 	}
 
-	private Optional<JSElement> iterateElement(final XSElementDeclaration elementDecl) {
-
+	private Optional<JSElement> iterateElement(final XSElementDeclaration elementDecl, int depth) {
+		if(depth>MAX_RECURSION_DEPTH) {
```

Comment:
is it possible to make this check optional? or to set the default value for MAX_RECURSION_DEPTH to a higher value?